libyul/AsmAnalysis.cpp

```diff
@@ -577,21 +577,21 @@ bool AsmAnalyzer::warnOnInstructions(evmasm::Instruction _instr, SourceLocation
 		_instr == evmasm::Instruction::RETURNDATACOPY ||
 		_instr == evmasm::Instruction::RETURNDATASIZE
 	) && !m_evmVersion.supportsReturndata())
-		errorForVM("only available for Byzantium-compatible");
+		errorForVM(7079_error, "only available for Byzantium-compatible");
```

Comment:
How is this supposed to work - should we re-use the error code? Or rather set all of them to ``0000_error`` and have the script assign random new ones? I'm not sure what's the long-term idea for the codes myself actually, just wondering.

Reply:
I've manually assigned new ones. I don't even know where this script lives.

What is more worrying there are no errors which indicates none of this is tested?

Reply:
I think the general idea is to set all new error codes to ``0000_error`` at first and then run ``scripts/error_codes.py --fix``, which will assign random ones (at least that's what the comment in `liblangutil/Exceptions.h` says) - and I think it ensures and there are tests that guarantee uniqueness at any given point in time, but not uniqueness relative to the entire history... but yeah, not sure.

Reply:
> I think the general idea is to set all new error codes to `0000_error` at first and then run `scripts/error_codes.py --fix`

Yes, indeed. For convenience, I just created PR #9300 to generate next random error code with the `--next` flag. So one could pick the number and bypass intermediate `0000_error`.

> I've manually assigned new ones.

In general, I would prefer
[a] either use a consistent and logical scheme for error codes
[b] or go for random numbers.

Since [a] does not seem to be practical, only [b] is left.

I rather dislike sequential error codes for similar things in one place while we do not do it in others.

> What is more worrying there are no errors which indicates none of this is tested?

Currently error codes are added to expectations of syntax tests. We might need to do it also for command line tests, Boost tests, etc.

So it could be that the errors are covered by tests, but we do not see it yet. At least the string `"only available for Constantinople-compatible"` appears in **InlineAssembly.cpp** tests and **SolidityNameAndTypeResolution.cpp** test.

BTW, PR #9266 suggests a possible way to track error coverage in Boost tests.

Reply:
> Since [a] does not seem to be practical, only [b] is left.

Yet somehow Rust managed to do [a] :wink:

> I rather dislike sequential error codes for similar things in one place while we do not do it in others.

As stated this PR was still not complete, because the more worrying fact is that isoltest does not detect the discrepancy.

Reply:
> > Since [a] does not seem to be practical, only [b] is left.
> 
> Yet somehow Rust managed to do [a] 😉

Oh, I meant "not practical in our case" :). In principle, it is doable. But if codes are not supposed to change, it might be painful to reserve ranges for "similar future errors".